### PR TITLE
Level 1 a more defensive code

### PR DESCRIPTION
**Pull request by danielpappa**
Added some assertions, created a log and changed the amount variable type of income from integer to double.

### CODE REVIEW

**Review by Xuan127 — APPROVED** [review]
LGTM

File: src/main/java/cashleh/commands/DeleteExpense.java [hunk 1]
```diff
@@ -1,53 +1,34 @@
...
-
     @Override
     public void execute() throws CashLehMissingTransactionException {
         try {
+            int numberOfEntriesBeforeDeletion = expenseStatement.getNumberOfExpenses();
             String expenseBeingDeleted = expenseStatement.getExpense(expenseIndex - 1).toString();
             expenseStatement.deleteExpense(expenseIndex - 1);
+            int numberOfEntriesAfterDeletion = expenseStatement.getNumberOfExpenses();
+            assert numberOfEntriesBeforeDeletion == numberOfEntriesAfterDeletion + 1;
             Ui.printMultipleText(new String[] {
```
Comment (BenedictChannn) [inline src/main/java/cashleh/commands/DeleteExpense.java:24]: Nice!

File: src/test/java/cashleh/transaction/ExpenseStatementTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    Expense testExpense = new Expense("monthly rent", 1200);
+
+    @Test
+    void getExpense() throws CashLehMissingTransactionException {
+        testStatement.addExpense(testExpense);
+        assertEquals(testStatement.getExpense(0), testExpense);
+    }
+
+    @Test
+    void getNumberOfEntries() {
```
Comment (BenedictChannn) [inline src/test/java/cashleh/transaction/ExpenseStatementTest.java]: Would it be better if we rename this to getNumberOfExpenses?

File: src/test/java/cashleh/transaction/ExpenseStatementTest.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        assertEquals(testStatement.getNumberOfExpenses(), 0);
+        testStatement.addExpense(testExpense);
+        assertEquals(testStatement.getNumberOfExpenses(), 1);
+        testStatement.addExpense(testExpense);
+        assertEquals(testStatement.getNumberOfExpenses(), 2);
+    }
+
+    @Test
+    void getSumOfEntries() {
```
Comment (BenedictChannn) [inline src/test/java/cashleh/transaction/ExpenseStatementTest.java]: Would it be better if we rename this to getSumOfExpenses

File: src/test/java/cashleh/transaction/IncomeStatementTest.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    Income testIncome = new Income("pocket money", 200);
+
+    @Test
+    void getIncome() throws CashLehMissingTransactionException {
+        testStatement.addIncome(testIncome);
+        assertEquals(testStatement.getIncome(0), testIncome);
+    }
+
+    @Test
+    void getNumberOfEntries() {
```
Comment (BenedictChannn) [inline src/test/java/cashleh/transaction/IncomeStatementTest.java]: Would renaming this to getNumberOfIncomes be better

File: src/main/java/cashleh/exceptions/CashLehException.java [hunk 5]
```diff
@@ -13,5 +13,14 @@ public class CashLehException extends Exception {
     public CashLehException(String message) {
         super(message);
     }
+    public static CashLehException invalidAddIncomeFormat() {
```
Comment (BenedictChannn) [inline src/main/java/cashleh/exceptions/CashLehException.java]: I'm not sure if this is needed or is in the right place to include this

**Review by BenedictChannn — APPROVED** [review]
LGTM

**Pull request merged**